src/rest-server/src/models/v2/job/k8s.js

```diff
@@ -292,7 +291,7 @@ const convertFrameworkDetail = async (framework) => {
   return detail;
 };
 
-const generateTaskRole = (taskRole, labels, config, userToken) => {
+const generateTaskRole = async (taskRole, labels, config) => {
```

Comment:
do not change it to `async`

Reply:
Curious about the reason, for performance or something?

---
In reply to: [352489791](https://github.com/microsoft/pai/pull/3944#discussion_r352489791) [](ancestors = 352489791)

Reply:
if there're 10 task roles, executing this async function 10 times sequentially will take a lot of time.
and all await calls in those executions are same, not related to task roles